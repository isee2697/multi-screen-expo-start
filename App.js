import React from 'react';
import { StatusBar, View } from 'react-native';
import { AppLoading, ScreenOrientation } from 'expo';
import { device, func, gStyle } from './src/constants';

// tab navigator
import TabNavigator from './src/navigation/TabNavigator';

class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      isLoading: true,
      theme: 'light'
    };

    // is tablet?
    if (device.isTablet) {
      ScreenOrientation.allowAsync(
        ScreenOrientation.Orientation.ALL_BUT_UPSIDE_DOWN
      );
    }

    this.updateTheme = this.updateTheme.bind(this);
  }

  updateTheme(themeType) {
    this.setState({
      theme: themeType
    });
  }

  render() {
    const { isLoading, theme } = this.state;

    if (isLoading) {
      return (
        <AppLoading
          onFinish={() => this.setState({ isLoading: false })}
          startAsync={func.loadAssetsAsync}
        />
      );
    }

    return (
      <React.Fragment>
        <StatusBar barStyle={device.iOS ? 'dark-content' : 'light-content'} />

        <TabNavigator
          screenProps={{
            updateTheme: this.updateTheme
          }}
          theme={theme}
        />
      </React.Fragment>
    );
  }
}

export default App;
