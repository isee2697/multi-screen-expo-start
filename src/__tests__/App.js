import React from 'react';
import { shallow } from 'enzyme';

import App from '../../App';

const app = shallow(<App />);

describe('<App />', () => {
  it('renders correctly', () => {
    expect(app).toMatchSnapshot();
  });
});

describe('<App /> State', () => {
  it('state: isLoading = true', () => {
    // initial state: isLoading
    expect(app.state('isLoading')).toEqual(true);
    expect(app).toMatchSnapshot();
  });

  it('state: isLoading = false', () => {
    // mock isLoading completed
    app.setState({
      isLoading: false
    });

    expect(app).toMatchSnapshot();
  });
});
