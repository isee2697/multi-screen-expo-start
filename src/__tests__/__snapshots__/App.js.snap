// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> State state: isLoading = false 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <StatusBar
    animated={false}
    barStyle="dark-content"
    showHideTransition="fade"
  />
  <NavigationContainer />
</Component>
`;

exports[`<App /> State state: isLoading = true 1`] = `
<AppLoading
  onFinish={[Function]}
  startAsync={[Function]}
/>
`;

exports[`<App /> renders correctly 1`] = `
<AppLoading
  onFinish={[Function]}
  startAsync={[Function]}
/>
`;
